### PR TITLE
refactor load_accounts_index_for_shrink for easier sharing

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem

Plan to remove duplicate code.
#### Summary of Changes
Refactor for easier sharing.

Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -4289,30 +4292,26 @@ impl AccountsDb {
...
-                        .lock()
-                        .unwrap()
-                        .append(&mut alive_accounts);
-                    unrefed_pubkeys_collect
-                        .lock()
-                        .unwrap()
-                        .push(unrefed_pubkeys);
-                    alive_total_collect.fetch_add(alive_total, Ordering::Relaxed);
-                });
+                stored_accounts
```
Comment (apfitzge) [inline runtime/src/accounts_db.rs:4295]: is there an additional follow-up PR relying on this one?
Repeated code-blocks here and at 3652

**Review by apfitzge — APPROVED** [review]
lgtm - love the change from iterating over chunk index to *actual* chunks

File: runtime/src/accounts_db.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs:4295]: yes
and it had more complicated references
this duplicate code will be removed

**Comment by jeffwashington** [comment]
ci passed with this commit as part of it:
#28457

**Pull request merged**